pkg/providers/cloudstack/config/secrets.yaml

```diff
@@ -0,0 +1,14 @@
+{{- range $profile := .Profiles}}
+apiVersion: v1
+kind: Secret
```

Comment:
I'm thinking these secrets should be immutable. See https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable for implementation details. The motivation here would be if a management cluster has hundreds of workload clusters, it doesn't really make sense for each cluster to have its own set of credentials. Instead, I think they should be shared across workload clusters, but immutable to prevent accidentally breaking other clusters.

Reply:
Workload clusters except the management cluster doesn't have capc controller so there is no need to have these secrets. Only management cluster and bootstrap cluster have capc controller therefore need the secrets.

Reply:
I think we cleared up the misunderstanding and agreed that EKS-A would create these secrets as mutable. However, it would not update the secret if it already existed, and it would instead log a warning that the secret contents have changed. If users want to update the secret contents then they can do so manually.